migration/migration/015-tracing-large-tags-fix.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE _ps_trace.tag DROP CONSTRAINT tag_key_value_id_key_id_key;
+CREATE UNIQUE INDEX tag_key_value_id_key_id_key ON _ps_trace.tag (key, _prom_ext.jsonb_digest(value)) INCLUDE (id, key_id);
```

Comment:
can this not have value in the INCLUDE either if the value is too large?

Reply:
unfortunately, the value cannot be in the INCLUDE either:

```
create table foo (bar text, baz int);
create index on foo using btree (baz) include (bar);
insert into foo (bar, baz) select string_agg(x::text, ''), 42 from generate_series(1, 10000) x;
ERROR:  54000: index row requires 38912 bytes, maximum size is 8191
LOCATION:  index_form_tuple, indextuple.c:191
```

Reply:
Yeah, keeping the value in the INCLUDE is not an option. As discussed on slack we have two options if we really want to claw back performance:

- Use a pair of partial indexes: one with `AND length(value::text) < 10000` the other with `AND length(value::text) >= 10000`, include `value` in the first one and make sure to transform every query into a union of the two. (@pgnickb suggestion)
-  Use a custom data type that can store the actual value if it's small enough instead or along with a hash. (@jgpruitt suggestion)

Reply:
it doesn't have to be a custom data type, but the concept is what's important